capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -115,4 +133,67 @@ public long upvotePost(HttpServletRequest request) {
     datastore.put(postToUpvote);
     return currentUpvotes;
   }
+
+  /**
+   * Sorts a list of entities based on the sort type passed in and returns the sorted list. If the
+   * sort type does not exist, an empty list is returned.
+   */
+  public List<Entity> sortEntities(String sortType, List<Entity> entities) {
+    Map<String, Comparator<Entity>> sorters = initializeSorters();
+    Comparator<Entity> sortMethod = sorters.get(sortType);
+    if (sortMethod == null) {
+      return Arrays.asList();
+    }
+
+    Collections.sort(entities, sortMethod);
+    return entities;
+  }
+
+  /** Returns a Map of all comparators. */
+  private Map<String, Comparator<Entity>> initializeSorters() {
```

Comment:
this can be static since it doesn't rely on any state of the class.

Reply:
It can't be static. Although sortByNew and sortByTop don't rely on class variables, sortByTrending relies on getUpvoteRatio which relies on the clock instance of the class. This would cause initializeSorters to reply on the clock instance as well. I could make sortByNew and sortByTop static but it would look inconsistent, in my opinion